OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -614,12 +614,12 @@ public override bool Tick(Actor self)
 
 				// Reserve the exit cell
 				mobile.SetPosition(self, cell, subCell);
-				mobile.SetVisualPosition(self, pos);
 
+				// mobile.SetVisualPosition(self, self.CenterPosition);
 				if (delay > 0)
 					QueueChild(new Wait(delay));
 
-				QueueChild(mobile.VisualMove(self, pos, self.World.Map.CenterOfSubCell(cell, subCell)));
+				QueueChild(mobile.VisualMove(self, self.CenterPosition, self.World.Map.CenterOfSubCell(cell, subCell)));
```

Comment:
We really do need to be using `pos` here and in `SetVisualPosition` above. The `SetPosition` call also updates the `CenterPosition`, which we need to reset back to the transport exit point. We will want to pass the center position into the ctor too, but this requires deeper plumbing changes that we should defer to Next + 1.